File: struts_spring_hibernateJHO/src/main/java/com/rococo/global/model/grocery/GroceryModel.java
/* -------------------------------------------------------------------------------- 
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * Copyright (C) RococoGlobal Technologies, Inc - All Rights Reserved 2013
 * -------------------------------------------------------------------------------- */
package com.rococo.global.model.grocery;

import java.io.Serializable;
import java.util.Date;

import javax.persistence.Column;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import javax.persistence.Table;
import javax.persistence.Entity;

/**
 * @author Anthony Encarnacion - Hibernate Integration
 */

@SuppressWarnings("serial")
@Entity
@Table(name = "t_item")
public class GroceryModel implements Serializable{

    @Id
    @GeneratedValue
    @Column(name="ID")
    private int id;
    
    @Column(name="date")
    private Date date;
    
    @Column(name="status")
    private boolean status;
    
    @Column(name="name")
    private String name;

    /**
     * @return item id
     */
    public int getId() {
        return id;
    }

    /**
     * @param id item id
     */
    public void setId(int id) {
        this.id = id;
    }

    /**
     * @return date of grocery
     */
    public Date getItemDate() {
        return date;
    }

    /**
     * @param date date of grocery
     */
    public void setItemDate(Date itemDate) {
        this.date = itemDate;
    }

    /**
     * @return status of item Purchased or Not
     */
    public boolean isStatus() {
        return status;
    }

    /**
     * @param status if item is purchased or not
     */
    public void setStatus(boolean status) {
        this.status = status;
    }

    /**
     * @return name of the item
     */
    public String getName() {
        return name;
    }

    /**
     * @param name of the item
     */
    public void setName(String name) {
        this.name = name;
    }
}

File: struts_spring_hibernateJHO/src/main/typescript/main/model/GroceryModel.ts
/** --------------------------------------------------------------------------------
 * Unauthorized copying of this file, via any medium is strictly prohibited
 * Proprietary and confidential
 * Copyright (C) Rococo Global Technologies, Inc - All Rights Reserved 2013
 * -------------------------------------------------------------------------------- */
///<reference path='../../libs/angularjs/angular.d.ts' />


/**
 * Created with JetBrains WebStorm.
 * User: Vincent Racaza
 * Date: 8/13/13
 * Time: 8:36 AM
 * Object used to store Grocery data
 */
module Model{

    export class GroceryList{
        groceries:Model.GroceryItems[];
        status: number;


        constructor(data:any){
            this.groceries = data.groceries;
            this.status = data.status;

        }
    }

    export class GroceryItems{
        date:string;
        isOpen:boolean;
        items:Model.Grocery[];


        constructor(data:any){
            this.date = data.date;
            this.isOpen = data.isOpen;
            this.items = data.items;
        }
    }

    export class Grocery{
        id:number;
        itemName:string;
        status:boolean;
        editable:boolean;

        constructor(data:any){
            this.id = data.id;
            this.itemName = data.itemName;
            this.status = data.status;
            this.editable = data.editable;
        }

    }
}